### PR TITLE
Add smaller and listOfSized utilities.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Found in lottery / walrus might be useful to others.
Bikeshed?

### CODE REVIEW

**Comment by tranma** [comment]
:+1:
think `smaller` was from icicle too

File: disorder-core/src/Disorder/Core/Gen.hs [hunk 1]
```diff
@@ -32,3 +34,12 @@ maybeGen g = sized $ \s ->
...
     (s, Just <$> resize (s `div` 2) g)]
+
+-- | Generate something smaller
+smaller :: Gen a -> Gen a
+smaller g =
+  sized $ \s -> resize (s `div` 2) g
+
+-- | Generate a list this big.
+listOfSized :: Gen a -> Int -> Gen [a]
+listOfSized gen n = take n <$> infiniteListOf gen
```
Comment (thumphries) [inline disorder-core/src/Disorder/Core/Gen.hs]: is this different to `vectorOf`?

Comment (tranma) [inline disorder-core/src/Disorder/Core/Gen.hs]: no, probably because vectorOf wasn't in QC < 2.8

Comment (thumphries) [inline disorder-core/src/Disorder/Core/Gen.hs]: ! the more you know

**Pull request merged**